external/fv3fit/fv3fit/sklearn/_random_forest.py

```diff
@@ -63,6 +63,7 @@ def __init__(
         output_variables: Iterable[str],
         hyperparameters: RandomForestHyperparameters,
     ):
+        super().__init__(sample_dim_name, input_variables, output_variables)
```

Comment:
Yes, I'm not a big fan of inheritance because it often leads to symptoms like this. We could remove the inheritance by making Predictor a purely abstract/protocol class and removing its init behavior, instead relying on each predictor to set its `self.input_variables` and `self.output_variables`.

While the inheritance is here, in theory `self._model_wrapper` can modify the input/output variables, and `predictor.__init__` could be updated to do more than just set `self.<things>`, so there isn't a safe way to remove one of these. I'd like to leave it if that's OK.

I do want to set removing init from Predictor as a future PR task. This also came up when talking about DenseModel, which now takes in both `hyperparameters` and `hyperparameters.input_variables` due to Predictor's requirements on its init signature. I can do it or I'd also be happy to give it to someone else who's interested.